sawyer_gazebo/src/arm_kinematics_interface.cpp

```diff
@@ -251,33 +251,71 @@ void ArmKinematicsInterface::publishGravityTorques()
   if (joint_state.get())
   {
     intera_core_msgs::SEAJointState gravity_torques;
+    KDL::JntArray jnt_pos, jnt_vel, jnt_eff, jnt_accelerations, jnt_gravity_model, jnt_gravity_only, jnt_zero;
     auto j_state = *joint_state.get();
     auto num_jnts = kinematic_chain_map_[tip_name_].chain.getNrOfJoints();
+    jnt_accelerations.resize(num_jnts);
```

Comment:
Set the number of joints in the constructor. The values will also be initialized to zero so there's no need to call SetToZero().

Reply:
I dug into the KDL code a bit. The underlining data in KDL::JntArray is a Eigen::VectorXd data which does not guarantee that values will be set to zero at initialization or during resizing. The KDL::JntArray code calls VectorXd.setToZero in the constructor but not in the resize function.

http://docs.ros.org/hydro/api/orocos_kdl/html/jntarray_8cpp_source.html

Reply:
Thanks @rethink-forrest . I'll rework all `KDL::JntArray` resizing code in this file.